cmd/traffic/cmd/agent/forwarder.go

```diff
@@ -157,8 +221,83 @@ func (f *Forwarder) ForwardConn(clientConn *net.TCPConn) {
 	for i := 0; i < 2; i++ {
 		select {
 		case <-ctx.Done():
-			return
 		case <-done:
 		}
 	}
```

Comment:
I almost posted a comment about how the original code was wrong, but that was me mis-interpretting the code.

The original code would have (IMO) been more clearly written as
```go
	// Wait for both sides to close the connection
	for numClosed := 0; numClosed < 2; {
		select {
		case <-ctx.Done():
			return
		case <-done:
			numClosed++
		}
	}
```

That is:
 1. In the "happy" path, we don't want to terminate until `<-done` has been read twice, or
 2. In the "sad" path, we want to bail early if `ctx.Done()` gets closed.
 
With that in mind, I'm not sure how it's correct to remove the bail-early `return`.

But I also don't see how it's correct to orphan the goroutines as it did when bailing early.  It seems to me that the reason `done` is a channel at all (instead of a `sync.WaitGroup`) is in order to support context cancellation.